docs/swarm/ingress.md

```diff
@@ -0,0 +1,119 @@
+<!--[metadata]>
++++
+title = "Use swarm mode routing mesh"
+description = "Use the routing mesh to publish services externally to a swarm"
+keywords = ["guide", "swarm mode", "swarm", "network", "ingress", "routing mesh"]
+[menu.main]
+identifier="ingress-guide"
+parent="engine_swarm"
+weight=17
++++
+<![end-metadata]-->
+
+# Use swarm mode routing mesh
+
+Docker Engine swarm mode makes it easy to publish ports for services to make
+them available outside the swarm. All nodes participate in an ingress
+**routing mesh**. The routing mesh enables each node in the swarm to accept
+connections on published ports for any service running in the swarm, even if
+there's no task running on the node. The internal load balancer routes all
+incoming requests on a published port to an available container.
+
+In order to use the ingress network in the swarm, you need to have the following
+ports open between the swarm nodes before you enable swarm mode:
+
+* Port `7946` TCP/UDP for container network discovery.
+* Port `4789` UDP for the container ingress network.
+
+You must also open the published port between the swarm nodes and any external
+resources, such as an external load balancer, that require access to the port.
+
+## Publish a port for a service
+
+Use the `--publish` flag to publish a port when you create a service:
+
+```bash
+$ docker service create \
+  --name <SERVICE-NAME> \
+  --publish <PUBLISHED-PORT>:<TARGET-PORT> \
+  <IMAGE>
+```
+
+The <TARGET-PORT> is the port on the task containers.
+The <PUBLISHED-PORT> is the port where the swarm makes the service available.
+
+For example, the following command publishes port 80 in the nginx container to
+port 8080 for any node in the swarm:
+
+```
+$ docker service create \
+  --name my-web \
+  --publish 8080:80 \
+  --replicas 2 \
+  nginx
+```
+
+When you access port 8080 on any node, the swarm load balancer routes your
+request to an active container.
+
+![service ingress image](images/ingress-routing-mesh.png)
+
+You can publish a port for an existing service using the following command:
+
+```bash
+$ docker service update \
+  --publish-add <PUBLISHED-PORT>:<TARGET-PORT> \
+  <SERVICE>
+```
+
+You can use `docker service inspect` to view the service's published port. For
+instance:
+
+```bash
+$ docker service inspect --format="{{.Endpoint.Spec.Ports}}" my-web
+
+[{ tcp 80 8080}]
+```
+
+The output shows the <TARGET-PORT> from the containers and the <PUBLISHED-PORT>
+where nodes listen for requests for the service.
+
+## Configure an external load balancer
+
+You can configure an external load balancer route requests to a swarm service.
+For example, you could configure a HAProxy service to balance requests to an
```

Comment:
s/could/can/ ?

Can you make `HAProxy` a link to http://www.haproxy.org ?

I'd replace "service" with something else (or remove it), because it may lead to think that HAProxy is running as a service in the swarm.